### PR TITLE
Feature/profile shoppinglist

### PR DESCRIPTION
**Pull request by HryshkoAnton**
BL for shopping list [1101](https://github.com/ita-social-projects/GreenCity/issues/1101)
BL for checkboxes [1166](https://github.com/ita-social-projects/GreenCity/issues/1166)

### CODE REVIEW

File: src/app/component/shared/shared.module.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ import {
 } from './components';
 import { MatDialogModule } from '@angular/material/dialog';
 import { MatCheckboxModule } from '@angular/material/checkbox';
+import {MatProgressSpinnerModule} from '@angular/material/progress-spinner';
```
Comment (Gusvarov) [inline src/app/component/shared/shared.module.ts:16]: Please, add spaces like in other imports

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.html [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+            </li>
+        </ul>
+    </div>
+    <ng-template #empty>
+        <p>You have no items in your Shopping List</p>
+    </ng-template>
+ </div>
+ <ng-template #loading>
+    <mat-spinner color="grey" diameter="70" class="loading"></mat-spinner>
+ </ng-template>
```
Comment (Gusvarov) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.html:24]: Please, add a newline at the end of file

Comment (HryshkoAnton) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.html]: - [ ]

File: src/app/component/shared/shared.module.ts (hunk 1 above)
Comment (HryshkoAnton) [reply src/app/component/shared/shared.module.ts]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.html [hunk 3]
```diff
@@ -1,11 +1,25 @@
...
+                ></mat-checkbox>
+            </li>
+        </ul>
+    </div>
+    <ng-template #empty>
+        <p class="noItems">You don't have any items in your Shopping List</p>
+    </ng-template>
+ </div>
+ <ng-template #loading>
+    <mat-spinner color="grey" diameter="70" class="loading"></mat-spinner>
```
Comment (RomanDeveloperAcc) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.html:23]: If you have more than 1 attribute, you should write them in separate rows, so please, rewrite it

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts [hunk 4]
```diff
@@ -1,19 +1,50 @@
...
 
 @Component({
   selector: 'app-shopping-list',
   templateUrl: './shopping-list.component.html',
   styleUrls: ['./shopping-list.component.scss']
 })
 export class ShoppingListComponent implements OnInit {
-  public shoppingList: Array<string> = ['Шоппер', 'Еко чашка', 'Бамбукова щітка'];
+  public shoppingList;
+  public profileSubscription;
```
Comment (RomanDeveloperAcc) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.ts:12]: Plese, add types to these variables

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts [hunk 5]
```diff
@@ -1,19 +1,50 @@
...
+  constructor(private profileService: ProfileService) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.getShoppingList();
+  }
+
+  public getShoppingList(): void {
+    this.profileSubscription = this.profileService.getShoppingList().subscribe(
+      (success: ShoppingList[]) => this.shoppingList = success,
```
Comment (RomanDeveloperAcc) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.ts:22]: Please, check if everything is OK with punctuation

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts [hunk 6]
```diff
@@ -1,19 +1,50 @@
...
+    );
+  }
 
-  public getShoppingList(): Array<string> {
-    return this.shoppingList.slice(0, 3);
+  public toggleDone(item): void {
+    this.profileService.toggleStatusOfShoppingItem(item)
+      .subscribe(
+        (success) => this.updateDataOnUi(item),
```
Comment (RomanDeveloperAcc) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.ts:29]: If you have only one assignment or function call, you shouldn't write it in separate rows. Please, rewrite it and check the punctuation

File: src/app/component/user/components/profile/profile-service/profile.service.ts [hunk 7]
```diff
@@ -28,6 +29,11 @@ export class ProfileService {
     return this.http.get<CardModel>(`${this.backEnd}facts/dayFact/2`);
   }
 
+  public getShoppingList(): Observable<ShoppingList[]> {
+    this.setUserId();
+    return this.http.get<ShoppingList[]>(`${this.backEnd}goals/shoppingList/${this.userId}/language/en`);
```
Comment (RomanDeveloperAcc) [inline src/app/component/user/components/profile/profile-service/profile.service.ts:34]: General: it is better to leave an empty line before the 'return' for more readability

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.scss [hunk 8]
```diff
@@ -52,6 +56,18 @@ h2 > span {
...
+
+.noItems {
+  font-family: $title-font;
+  font-style: normal;
+  font-weight: 600;
+  font-size: 14px;
+  line-height: 17px;
+  margin-top: 30px;
+  text-align: center;
+  color: #bebebe;
```
Comment (mr-pavlenko) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.scss:68]: Colors should have the same writing style.

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts [hunk 9]
```diff
@@ -1,19 +1,50 @@
...
+      );
   }
 
+  private updateDataOnUi(item): void {
+
+    const index = this.shoppingList.findIndex(shoppingItem => shoppingItem.goalId === item.goalId);
+    const newItemStatus = item.status === 'ACTIVE' ? 'DONE' : 'ACTIVE';
```
Comment (mr-pavlenko) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.ts:36]: status should be on constant

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.html (hunk 3 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.html]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts (hunk 4 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.ts]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts (hunk 5 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.ts]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts (hunk 6 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.ts]: done

File: src/app/component/user/components/profile/profile-service/profile.service.ts (hunk 7 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/profile-service/profile.service.ts]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.scss (hunk 8 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.scss]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.ts (hunk 9 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.ts]: done

File: src/app/component/user/components/profile/profile-service/profile.service.ts [hunk 10]
```diff
@@ -48,4 +58,13 @@ export class ProfileService {
   public getUserFriends(): Observable<object> {
     return this.http.get<object>(`${this.backEnd}user/${this.userId}/sixUserFriends/`);
   }
+
+  public toggleStatusOfShoppingItem(item): Observable<object[]> {
+    const { status, goalId } = item;
+    const newStatus = status === 'DONE' ? false : true;
+    const params = new HttpParams()
+      .set('goalId', goalId)
+      .set('status', newStatus.toString());
```
Comment (RomanDeveloperAcc) [inline src/app/component/user/components/profile/profile-service/profile.service.ts:67]: As it was mentioned, it is better to leave an empty line before the 'return'

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.html [hunk 11]
```diff
@@ -1,11 +1,24 @@
...
-            <mat-checkbox class="checkbox" color="primary"></mat-checkbox>
-        </li>
-    </ul>
 </div>
+<ng-template #empty>
+    <p class="noItems">You don't have any items in your Shopping List</p>
+</ng-template>
+<ng-template #loading>
+    <mat-spinner 
+        color="grey"
```
Comment (RomanDeveloperAcc) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.html:20]: Bad alignment: first attribute in the same row, where the tag is, the others should be in separate row, under the first. Check any .html file in the project to understand it better

File: src/app/component/user/components/profile/profile-service/profile.service.ts (hunk 10 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/profile-service/profile.service.ts:67]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.html (hunk 11 above)
Comment (HryshkoAnton) [reply src/app/component/user/components/profile/shopping-list/shopping-list.component.html]: done

File: src/app/component/user/components/profile/shopping-list/shopping-list.component.html [hunk 12]
```diff
@@ -1,11 +1,26 @@
...
 </div>
+<ng-template #empty>
+    <p class="noItems">You don't have any items in your Shopping List</p>
+</ng-template>
+<ng-template #loading>
+    <mat-spinner color="grey" 
+                 diameter="70" 
+                 class="loading"
+    ></mat-spinner>
+</ng-template>
```
Comment (mplesha) [inline src/app/component/user/components/profile/shopping-list/shopping-list.component.html:26]: fix it

File: src/app/component/user/components/profile/profile-service/profile.service.ts [hunk 13]
```diff
@@ -48,4 +58,14 @@ export class ProfileService {
   public getUserFriends(): Observable<object> {
     return this.http.get<object>(`${this.backEnd}user/${this.userId}/sixUserFriends/`);
   }
+
+  public toggleStatusOfShoppingItem(item): Observable<object[]> {
+    const { status, goalId } = item;
+    const newStatus = status === 'DONE' ? false : true;
```
Comment (mplesha) [inline src/app/component/user/components/profile/profile-service/profile.service.ts:64]: fix it

**Pull request merged**